pdr_backend/pdr_dashboard/util/data.py

```diff
@@ -126,3 +131,95 @@ def find_with_key_value(
             return obj
 
     return None
+
+
+def get_feeds_stat_with_contract(
+    contract: str, feed_stats: List[Dict[str, Any]]
+) -> Dict[str, Union[float, int]]:
+    result = find_with_key_value(feed_stats, "contract", contract)
+
+    if result:
+        return {
+            "avg_accuracy": float(result["avg_accuracy"]),
+            "avg_stake_per_epoch_(OCEAN)": float(result["avg_stake"]),
+            "volume_(OCEAN)": float(result["volume"]),
+        }
+
+    return {
+        "avg_accuracy": 0,
+        "avg_stake_per_epoch_(OCEAN)": 0,
+        "volume_(OCEAN)": 0,
+    }
+
+
+def get_feeds_subscription_stat_with_contract(
+    contract: str, feed_subcription_stats: List[Dict[str, Any]]
+) -> Dict[str, Union[float, int, str]]:
+    result = find_with_key_value(feed_subcription_stats, "contract", contract)
+
+    if result:
+        sales_str = f"{result['sales']}"
+
+        df_buy_count_str = (
+            f"{result['df_buy_count']}-DF" if result["df_buy_count"] > 0 else ""
+        )
+        ws_buy_count_str = (
+            f"{result['ws_buy_count']}-WS" if result["ws_buy_count"] > 0 else ""
+        )
+
+        if df_buy_count_str or ws_buy_count_str:
+            counts_str = "_".join(filter(None, [df_buy_count_str, ws_buy_count_str]))
+
+            if counts_str:
+                sales_str += f"_{counts_str}"
+
+        return {
+            "price_(OCEAN)": result["price"],
+            "sales": sales_str,
+            "sales_raw": result["sales"],
+            "sales_revenue_(OCEAN)": result["sales_revenue"],
+        }
+
+    return {
+        "price_(OCEAN)": 0,
+        "sales": 0,
+        "sales_raw": 0,
+        "sales_revenue_(OCEAN)": 0,
+    }
+
+
+def get_feeds_data_for_feeds_table(
```

Comment:
this is an excellent candidate of why presentation should go into db getter: it's just renaming some columns and presenting them differently. (Not part of the main code review, but discussed for a further iteration)